src/app/gui/map/mapservice.js

```diff
@@ -1844,84 +1844,70 @@ proto._setupAllLayers = function() {
   });
 };
 
-//SETUP MAPLAYERS
+// SETUP MAPLAYERS
 proto._setupMapLayers = function() {
   // get all geolayers exclude baselayers and eventually vector layers
-  const layers = getMapLayersByFilter({
-    BASELAYER: false,
-    VECTORLAYER: false
-  });
+  const layers = getMapLayersByFilter({ BASELAYER: false, VECTORLAYER: false });
+
   this._setMapProjectionToLayers(layers);
-  //group layer by mutilayer (multilayer property of layer on project configuration)
-  // nee to split time series to group to speed up eventualli time seriesries loading of single layer
-  let qtimeseries_multilayerid_split_values = {};
-  const multiLayers = _.groupBy(layers, layer => {
-    let multiLayerId = layer.getMultiLayerId();
-    if (layer.isQtimeseries()) {
-      qtimeseries_multilayerid_split_values[multiLayerId] = qtimeseries_multilayerid_split_values[multiLayerId] === undefined ? 0 : qtimeseries_multilayerid_split_values[multiLayerId] + 1;
-      multiLayerId = `${multiLayerId}_${qtimeseries_multilayerid_split_values[multiLayerId]}`;
-    } else multiLayerId = qtimeseries_multilayerid_split_values[multiLayerId] === undefined ?
-      multiLayerId : `${multiLayerId}_${qtimeseries_multilayerid_split_values[multiLayerId] + 1}`;
-    return multiLayerId;
-  });
-  qtimeseries_multilayerid_split_values = null; // delete to garbage collector
+
+  let cache     = {};
   let mapLayers = [];
-  Object.entries(multiLayers).forEach(([id, layers]) => {
-    const multilayerId = `layer_${id}`;
-    let mapLayer;
-    const layer = layers[0] || [];
-    if (layers.length === 1) {
-      mapLayer = layer.getMapLayer({
-        id: multilayerId,
-        projection: this.getProjection(),
-        /**
-         * @since 3.7.11
-         */
-        format: layer.getSource() && layer.getSource().format
-      }, {});
-      mapLayer.addLayer(layer);
-      mapLayers.push(mapLayer)
-    } else {
-      mapLayer = layer.getMapLayer({
-        id: multilayerId,
-        projection: this.getProjection()
-      }, this.layersExtraParams);
-      layers.reverse().forEach(sub_layer => mapLayer.addLayer(sub_layer));
+
+  Object
+    .entries(
+      // Group layers by multilayer property (from project config)
+      // to speed up "qtimeseriesries" loading for single layers
+      _.groupBy(layers, layer => {
+        let id = layer.getMultiLayerId();
+        if (layer.isQtimeseries()) {
+          cache[id] = undefined === cache[id] ? 0 : cache[id] + 1;
+          return `${id}_${cache[id]}`;
+        }
+        return id = undefined === cache[id] ? id : `${id}_${cache[id] + 1}`;
+      })
+    )
+    .forEach(([id, layers]) => {
+      const layer    = layers[0] || [];
+      const mapLayer = layer.getMapLayer(
+        {
+          id: `layer_${id}`,
+          projection: this.getProjection(),
+          /** @since 3.7.11 */
+          format: (1 === layers.length && layer.isExternalWMS()) ? layer.getSource() && layer.getSource().format : null
```

Comment:
@volterra79 do you think it's possible to consolidate the `layer.isExternalWMS()` check in here? [src/app/core/layers/map/wmslayer.js#L91](https://github.com/g3w-suite/g3w-client/pull/531/files#diff-fc9f6aaf4a8d83ea2d7df3810f609900bd5e7877bc109a07d85af4da82210a66)

Reply:
@Raruto can you check https://github.com/g3w-suite/g3w-client/pull/531/commits/7e3fbf5e211b926eb9cc8af00c0a722677369fc0 commit